src/python/pants/backend/python/macros/poetry_requirements.py

```diff
@@ -167,66 +199,106 @@ def non_pants_project_abs_path(self, path: str) -> Path | None:
             )
 
 
+def produce_match(sep: str, feat: Any) -> str:
+    return f"{sep}{feat}" if feat else ""
+
+
+def add_markers(base: str, attributes: PyprojectAttr, fp) -> str:
+    markers_lookup = produce_match("", attributes.get("markers"))
+    python_lookup = parse_python_constraint(attributes.get("python"), fp)
+
+    # Python constraints are passed as a `python_version` environment marker; if we have multiple
+    # markers, we evaluate them as one whole, and then AND with the new marker for the Python constraint.
+    # E.g. (marker1 AND marker2 OR marker3...) AND (python_version)
+    # rather than (marker1 AND marker2 OR marker3 AND python_version)
+    if not markers_lookup and not python_lookup:
+        return base
+
+    result = base
+    if markers_lookup or python_lookup:
+        result += ";"
+        result += "("
```

Comment:
You already know the condition is true, given line 214:

```suggestion
    result = f"{base} ; ("
```

(Not sure if those suggested white spaces are legal or not)

Reply:
The whitespaces just get removed through Requirement.parse() anyway so I had it without spaces; probably best to remove to avoid updating tests that deal with strings.

Reply:
Okay, makes sense. I'll merge this PR now, given that it's green and this suggestion isn't a very big deal. If you happen to touch this part of the codebase soon, it'd be good to still make this cleanup.